lang/c++/impl/NodeImpl.cc

```diff
@@ -125,54 +125,60 @@ NodeUnion::resolve(const Node &reader) const
     return match;
 }
 
-SchemaResolution 
+SchemaResolution
 NodeFixed::resolve(const Node &reader) const
 {
     if(reader.type() == AVRO_FIXED) {
         return (
                 (reader.fixedSize() == fixedSize()) &&
-                (reader.name() == name()) 
-            ) ? 
+                (reader.name() == name())
+            ) ?
             RESOLVE_MATCH : RESOLVE_NO_MATCH;
     }
     return furtherResolution(reader);
 }
 
-SchemaResolution 
+SchemaResolution
 NodeSymbolic::resolve(const Node &reader) const
 {
     const NodePtr &node = leafAt(0);
     return node->resolve(reader);
 }
 
-// Wrap an indentation in a struct for ostream operator<< 
-struct indent { 
+// Wrap an indentation in a struct for ostream operator<<
+struct indent {
     indent(int depth) :
         d(depth)
     { }
-    int d; 
+    int d;
 };
 
 /// ostream operator for indent
 std::ostream& operator <<(std::ostream &os, indent x)
 {
     static const std::string spaces("    ");
     while(x.d--) {
-        os << spaces; 
+        os << spaces;
     }
     return os;
 }
 
-void 
+void
 NodePrimitive::printJson(std::ostream &os, int depth) const
 {
     os << '\"' << type() << '\"';
+    if (getDoc().size()) {
+        os << ",\n" << indent(depth) << "\"doc\": \"" << getDoc() << "\"";
```

Comment:
Once we add escape() function in https://github.com/apache/avro/pull/241/files we should wrap getDoc() in escape().

Reply:
ack